src/rules/ec2/EC2IMDSv2.ts

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+import { parse } from 'path';
+import { CfnResource, Stack } from 'aws-cdk-lib';
+import {
+  CfnAutoScalingGroup,
+  CfnLaunchConfiguration,
+} from 'aws-cdk-lib/aws-autoscaling';
+import { CfnInstance, CfnLaunchTemplate } from 'aws-cdk-lib/aws-ec2';
+import { NagRuleCompliance, NagRules } from '../../nag-rules';
+
+// CfnLaunchConfiguration
+
+// Check to see if the EC2 Istance is configured with a launch profile
+//Ec2imdsv2
+
+/**
+ * Determine if the EC2 Instance has a launch profile
+ * @param node the CfnResource to check
+ */
+export default Object.defineProperty(
+  (node: CfnResource): NagRuleCompliance => {
+    if (node instanceof CfnInstance) {
+      // to use imdsv2 there must be a launchtemplate.
```

Comment:
Let's remove the inline comments for the code

Reply:
Ok, if thats your style..   I find single line comments really helpful to provide the intention of whats happening.  But now removed.